cmd/krew/cmd/search.go

```diff
@@ -43,58 +44,91 @@ Examples:
   To fuzzy search plugins with a keyword:
     kubectl krew search KEYWORD`,
 	RunE: func(cmd *cobra.Command, args []string) error {
-		plugins, err := indexscanner.LoadPluginListFromFS(paths.IndexPluginsPath(constants.DefaultIndexName))
-		if err != nil {
-			return errors.Wrap(err, "failed to load the list of plugins from the index")
+		indexes := []indexoperations.Index{
+			{
+				Name: constants.DefaultIndexName,
+				URL:  constants.IndexURI, // unused here but providing for completeness
+			},
 		}
-		names := make([]string, len(plugins))
-		pluginMap := make(map[string]index.Plugin, len(plugins))
-		for i, p := range plugins {
-			names[i] = p.Name
-			pluginMap[p.Name] = p
+		if os.Getenv(constants.EnableMultiIndexSwitch) != "" {
+			out, err := indexoperations.ListIndexes(paths)
+			if err != nil {
+				return errors.Wrapf(err, "failed to list plugin indexes available")
+			}
+			indexes = out
 		}
 
+		klog.V(3).Infof("found %d indexes", len(indexes))
+
+		var plugins []pluginEntry
+		for _, idx := range indexes {
+			ps, err := indexscanner.LoadPluginListFromFS(paths.IndexPluginsPath(constants.DefaultIndexName))
```

Comment:
Should this be `paths.IndexPluginsPath(idx.Name)`? As it is now it seems like it will only load plugins from the default index even if `X_KREW_ENABLE_MULTI_INDEX` is set.

Reply:
Nice catch! Why the hell are tests not catching this I’ll take a look.

Reply:
It's because we copy the default index into the custom index

Reply:
Ah, right. I guess it cannot be helped for now :)